api/types/apitypes.go

```diff
@@ -285,8 +285,15 @@ type BlockDataBasic struct {
 	StakeDiff  float64 `json:"sdiff,omitemtpy"`
 	Time       int64   `json:"time,omitemtpy"`
 	NumTx      uint32  `json:"txlength,omitempty"`
-	//TicketPoolInfo
-	PoolInfo TicketPoolInfo `json:"ticket_pool,omitempty"`
+	// TicketPoolInfo may be nil for side chain blocks.
+	PoolInfo *TicketPoolInfo `json:"ticket_pool,omitempty"`
+}
+
+// NewBlockDataBasic constructs a *BlockDataBasic with all fields allocated.
+func NewBlockDataBasic() *BlockDataBasic {
```

Comment:
"with all fields allocated" is confusing as you are just initializing `PoolInfo` field.

Reply:
"with valid pointer fields" feels better unless you've got a better option.